src/sources/kubernetes_logs/mod.rs

```diff
@@ -353,8 +394,19 @@ impl Source {
                 byte_size,
                 pod_name: file_info.as_ref().map(|info| info.pod_name),
             });
+
             if file_info.is_none() {
                 emit!(KubernetesLogsEventAnnotationFailed { event: &event });
+            } else {
+                let namespace = file_info.as_ref().map(|info| info.pod_namespace);
+
+                if let Some(name) = namespace {
+                    let ns_info = ns_annotator.annotate(&mut event, name);
+
+                    if ns_info.is_none() {
+                        emit!(KubernetesLogsEventNamespaceAnnotationFailed { event: &event });
```

Comment:
I'm gonna guess users are going to ask "why" if they see this error. Maybe we could make `annotate` return a `Result` instead and codify the errors to include with the event here?

Reply:
Should we also update the pod annotator, which also doesn't return any information as to "why"? I'm not sure we have too much context other than "we didn't find an entry for this"

Reply:
Ah, yeah I see that. Maybe we should just report `NamespaceMetadataNotFound` or something similar then?

We could do the same for pods.

Reply:
Do we need to note a breaking change if the internal events are changed?

Reply:
Yeah, that's a good question. My inclination is yes albeit very minor. We intend to define a policy around this this quarter.

Reply:
Given this is matching the current pod annotations, I'm fine with punting and creating a separate issue for updating them both.

Reply:
Hokay, I'll update the internal events this evening/tomorrow morning